File: samples/client/wordnik-api/python/tests/AccountApiTest.py
#!/usr/bin/env python

import sys
import unittest
import urllib2
import json

from BaseApiTest import BaseApiTest

sys.path = ['./'] + sys.path
from wordnik import *


class AccountApiTest(BaseApiTest):

    def setUp(self):
        super(AccountApiTest, self).setUp()
        self.authToken = self.accountApi.authenticate(self.username,
                                                      self.password).token

    def testAuthenticate(self):
        res = self.accountApi.authenticate(self.username, self.password)
        assert res, 'null authenticate result'
        assert res.token, 'invalid authentication token'
        assert res.userId != 0, 'userId was 0'
        assert res.userSignature, 'invalid userSignature'

    def testAuthenticatePost(self):
        res = self.accountApi.authenticatePost(self.username, self.password)
        assert res, 'null authenticate result'
        assert res.token, 'invalid authentication token'
        assert res.userId != 0, 'userId was 0'
        assert res.userSignature, 'invalid userSignature'

    def testGetWordListsForLoggedInUser(self):
        res = self.accountApi.getWordListsForLoggedInUser(self.authToken)
        assert res, 'null getWordListsForLoggedInUser result'
        assert len(res) != 0, 'number of lists shouldn\'t be 0'

    def testGetApiTokenStatus(self):
        res = self.accountApi.getApiTokenStatus()
        assert res, 'null getApiTokenStatus result'
        assert res.valid, 'token status not valid'
        assert res.remainingCalls != 0, 'remainingCalls shouldn\'t be 0'

    def testGetLoggedInUser(self):
        res = self.accountApi.getLoggedInUser(self.authToken)
        assert res, 'null getLoggedInUser result'
        assert res.id != 0, 'if shouldn\'t be 0'
        assert res.username == self.username, 'username was incorrect'
        assert res.status == 0, 'user status should be 0'
        assert res.email, 'email shouldn\'t be null'


if __name__ == "__main__":
    unittest.main()

File: samples/client/wordnik-api/php/tests/AccountApiTest.php
<?php

require_once 'BaseApiTest.php';

class AccountApiTest extends BaseApiTest {

  public function setUp() {
    parent::setUp();
    $this->authToken = $this->accountApi->authenticate($this->username,
                                                      $this->password)->token;
    date_default_timezone_set('America/Los_Angeles');
  }

  public function testAuthenticate() {
    $res = $this->accountApi->authenticate($this->username,
                                           $this->password);
    $this->assertObjectHasAttribute('token', $res);
    $this->assertNotEquals(0, $res->userId);
    $this->assertObjectHasAttribute('userSignature', $res);
  }

  public function testAuthenticatePost() {
    $res = $this->accountApi->authenticatePost($this->username,
                                           $this->password);
    $this->assertObjectHasAttribute('token', $res);
    $this->assertNotEquals(0, $res->userId);
    $this->assertObjectHasAttribute('userSignature', $res);
  }

  public function testGetWordListsForLoggedInUser() {
    $res = $this->accountApi->getWordListsForLoggedInUser($this->authToken);
    $this->assertNotEquals(0, count($res));
  }

  public function testGetApiTokenStatus() {
    $res = $this->accountApi->getApiTokenStatus();
    $this->assertObjectHasAttribute('valid', $res);
    $this->assertNotEquals(0, count($res->remainingCalls));
  }

  public function testGetLoggedInUser() {
    $res = $this->accountApi->getLoggedInUser($this->authToken);
    $this->assertNotEquals(0, $res->id);
    $this->assertEquals($this->username, $res->username);
    $this->assertEquals(0, $res->status);
    $this->assertNotEquals(null, $res->email);
  }


}
?>
